src/qtxdg/xdgmenureader.cpp

```diff
@@ -357,8 +360,9 @@ void XdgMenuReader::processDefaultDirectoryDirsTag(QDomElement& element)
     QStringList dirs = XdgDirs::dataDirs();
     dirs.prepend(XdgDirs::dataHome(false));
 
-    for (const QString &dir : qAsConst(dirs))
-        addDirTag(element, QLatin1String("DirectoryDir"), dir + QLatin1String("/desktop-directories/"));
+    int n = dirs.size();
+    for (int i = 0; i < n; ++i)
```

Comment:
I would rather use reverse iterators ...

Reply:
I don't think there would be a gain; would be?

Reply:
I'm not sure. Just the computation of ```n - i - 1``` wouldn't be needed in each iteration.

Reply:
It's so cheap, perhaps the cheapest.

There are opposite theoretical or aesthetic opinions. I don't think we need to worry about them.

Reply:
An example, regarding `std::vector`: https://itecnote.com/tecnote/c-speed-accessing-a-stdvector-by-iterator-vs-by-operator-index/

Reply:
> It's so cheap, perhaps the cheapest.

It may be the cheapest thing here, but it is completely unneeded and avoidable.

> An example, regarding std::vector: https://itecnote.com/tecnote/c-speed-accessing-a-stdvector-by-iterator-vs-by-operator-index/

I'm not arguing about iterators vs. `[]`, you're right, that in most cases that's just a personal preference.

Maybe I was bit misleading in the beginning with saying to rather use iterators. But using the iterators here would implicitly prevent to use the suboptimal index computation.

Anyway, even with the index loop I would suggest to use `for (int i = dirs.size() - 1; i >= 0; --i)`.

Reply:
We're only talking about different tastes here. The difference is zero in practice.